packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/publisher.ts

```diff
@@ -0,0 +1,169 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+// Keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * An interface that will be implemented by all the resources that can be published events to.
+ */
+export interface IPublisher {
+  publish(taskDefinition: ecs.TaskDefinition): void;
+
+  envVarKey(): string;
+
+  envVarValue(): string;
+}
```

Comment:
This interface doesn't look quite right to me...It's difficult to related `envVarKey` and `envVarValue` with `IPublisher`

Reply:
Yes I agree, these aren't related to the actual 'publish' functionality that we want to provide the users. But if these aren't added to the interface, we will end up doing `instanceof` type checks later in the code and we wanted to avoid such logic and find the most common interface. That's why `envVarKey` and `envVarValue` were added to the interface itself.